### ISSUE TITLE
[Kotlin/Native, iOS] Unexpected receiver type: noname of RuntimeException

### ISSUE THREAD

**Issue by inoutch** [JetBrains/kotlin-native#3100]
I am developing an iOS application using my library and it. Today, an error will be thrown if I write the following code:
```
// enabledExtensionName is List<String> that have 2 strings.
// this line is executed from viewDidLoad()
~
enabledExtensionNames.toCStringArray(scope)
~
```

The errors:
```
Uncaught Kotlin exception: kotlin.RuntimeException: Unexpected receiver type: noname
        at 0   kotlin-example                      0x0000000100a8bb58 kfun:kotlin.Throwable.<init>(kotlin.String?)kotlin.Throwable + 24
        at 1   kotlin-example                      0x0000000100a8bb1c kfun:kotlin.Exception.<init>(kotlin.String?)kotlin.Exception + 20
        at 2   kotlin-example                      0x0000000100a8baf4 kfun:kotlin.RuntimeException.<init>(kotlin.String?)kotlin.RuntimeException + 20
        at 3   kotlin-example                      0x0000000100a8b9f8 kfun:kotlin.native.internal.ThrowInvalidReceiverTypeException(kotlin.reflect.KClass<#STAR>)kotlin.Nothing + 184
        at 4   kotlin-example                      0x0000000100aaa41c kfun:kotlin.collections.collectionSizeOrDefault$stdlib@kotlin.collections.Iterable<#GENERIC>.(kotlin.Int)Generic + 184
        at 5   kotlin-example                      0x0000000100bee590 kfun:kotlinx.cinterop.toCStringArray@kotlin.collections.List<kotlin.String>.(kotlinx.cinterop.AutofreeScope)kotlinx.cinterop.CPointer<kotlinx.cinterop.CPointerVarOf<kotlinx.cinterop.CPointer<kotlinx.cinterop.ByteVarOf<kotlin.Byte>>>> + 92
        ~
```

The above code is implemented on an external library built by Kotlin MPP, and it is imported.
Oddly enough, if I write similar code before this one, the above error occur on and off.
For example, if the following code is written in main.kt, the code in the above library will be passed.
```
// main.kt for iOS
memScoped {
    listOf("1", "2").toCStringArray(this) // <<<- THIS Meaningless code for pass
    autoreleasepool {
        UIApplicationMain(args.size, args.toCStringArray(this), null, NSStringFromClass(AppDelegate))
    }
}
```
However, other error is thrown
```
Uncaught Kotlin exception: kotlin.RuntimeException: Unexpected receiver type: kotlin.collections.ArrayList
        at 0   kotlin-example                      0x0000000104633738 kfun:kotlin.Throwable.<init>(kotlin.String?)kotlin.Throwable + 24
        at 1   kotlin-example                      0x00000001046336fc kfun:kotlin.Exception.<init>(kotlin.String?)kotlin.Exception + 20
        at 2   kotlin-example                      0x00000001046336d4 kfun:kotlin.RuntimeException.<init>(kotlin.String?)kotlin.RuntimeException + 20
        at 3   kotlin-example                      0x00000001046335d8 kfun:kotlin.native.internal.ThrowInvalidReceiverTypeException(kotlin.reflect.KClass<#STAR>)kotlin.Nothing + 184
        at 4   kotlin-example                      0x000000010464f1f4 kfun:kotlin.collections.collectionSizeOrDefault$stdlib@kotlin.collections.Iterable<#GENERIC>.(kotlin.Int)Generic + 260
        ~
```
Where the error applies
```
// deviceQueueFamilyProperties is List<MyObject>
deviceQueueFamilyProperties.indexOfFirst {
    ~
}
```
I think this problem did not occur before Kotlin 1.3.20.
Is this a bug? or is there my any failure in memory management?

**Comment by inoutch** [comment]
I found that this problem is caused by applying the kotlin-serialization plugin and adding to dependencies. If you apply this plugin and call toCStringArray method in ViewController, an error occurs.

kotlin: v1.3.40
kotlin serialization: v0.11.1

**Comment by homuroll** [comment]
Should be fixed in 1.3.41
